frontend/src/app/search/highlight-selector.component.html

```diff
@@ -1,6 +1,26 @@
 <div class="contextbox">
-    <h2 class="subtitle">Context (no. of characters):</h2>
-    <p class="control" iaBalloon="Leave blank or set to 0 to turn off highlighting" iaBalloonLength="medium" iaBalloonPosition="right">
-        <input class="input" type="number" (change)="updateHighlightSize($event)">
-    </p>
+    <div>
+        <h2 class="subtitle highlights-subtitle">Toggle highlights:</h2>
+    </div>
+    <div class="column is-narrow">
+        <div class="field has-addons">
+            <div class="extended-selector">
+                <div *ngIf="highlight!=0"  class="field has-addons">
+                    <div class="control" iaBalloon="Increase/decrease preview size" iaBalloonLength="medium" iaBalloonPosition="right">
+                        <button class="button is-light" (click)="updateHighlightSize('more')">
+                            <i class="fa fa-plus" aria-hidden="true"></i>
+                        </button>
+                        <button class="button is-light" (click)="updateHighlightSize('less')">
+                            <i class="fa fa-minus" aria-hidden="true"></i>
+                        </button>
+                    </div>
+                </div>
+            </div>
+            <div class="control">
+                <button class="button highlight-toggle" [class.is-active]="highlight!=0" [disabled]="queryModel.queryText==null||queryModel.queryText==''" (click)="updateHighlightSize(highlight!=0 ? 'off' : 'on')">
```

Comment:
I think the query text could also be `undefined`, not sure. `[disabled]=!queryModel.queryText` would the the quicker way to do this check, I think, but you could also add a `hasQueryText` property to the query model.

If the highlight is disabled, can you add a balloon explaining why?